index.html

```diff
@@ -650,21 +654,31 @@ <h3>The <dfn data-dfn-for="GlobalEventHandlers" data-dfn-type="event">lostpointe
+                <p>These events MUST be of type <code>PointerEvent</code>, and are subject to the additional requirements mentioned in the rest of this section.</p>
+                <section>
+                  <h4>Attributes (other than coordinates)</h4>
+                  <p>For these events, all <code>PointerEvent</code> specific attributes (defined in this spec) other than <code>pointerId</code> and <code>pointerType</code> MUST have their default values. In addition:</p>
+                  <ul>
+                    <li>If the events are generated by a pointing device, their <code>pointerId</code> and <code>pointerType</code> MUST be the same as the PointerEvents that caused these events.</li>
+                    <li>If the events are generated by a non-pointing device (such as voice recognition software or a keyboard interaction), <code>pointerId</code> MUST be <code>-1</code> and <code>pointerType</code> MUST be an empty string.</li>
+                  </ul>
+                </section>
+                <section>
+                  <h4>Event coordinates</h4>
+                  <p>As noted in {{PointerEvent}}, [[[CSSOM-VIEW]]] proposes to redefine the various coordinate properties (<code>screenX</code>, <code>screenY</code>, <code>pageX</code>, <code>pageY</code>, <code>clientX</code>,
+                      <code>clientY</code>, <code>x</code>, <code>y</code>, <code>offsetX</code>, <code>offsetY</code>) as <code>double</code>, to allow for fractional coordinates.
+                      However, this change — when applied only to {{PointerEvent}}, but not to regular {{MouseEvent}} — has proven to lead to web compatibility issues with legacy code
+                      in the case of <code>click</code>, <code>auxclick</code>, and <code>contextmenu</code>. For this reason, user agents that have implemented the proposed
+                      change in [[[CSSOM-VIEW]]] only for {{PointerEvent}} MUST convert the various coordinate properties for the <code>click</code>, <code>auxclick</code>, and <code>contextmenu</code>
+                      to <code>long</code> values (as defined in the original [[[UIEVENTS]]]) using <a data-cite="ECMASCRIPT#sec-math.floor">Math.floor</a> [[ECMASCRIPT]].</p>
+                </section>
+                <section>
+                  <h4>Event dispatch</h4>
+                  <p>Let <var>e</var> be any <code>click</code>, <code>auxclick</code> or <code>contextmenu</code> event, and <var>t</var> be the user interaction event that caused the firing of <var>e</var>. The type of <var>t</var> depends on native platform convention: if <var>e</var> is a <code>contextmenu</code> event, <var>t</var> is either a {{GlobalEventHandlers/pointerdown}} or a {{GlobalEventHandlers/pointerup}} event; if <var>e</var> is a <code>click</code> or <code>auxclick</code> event, <var>t</var> is a {{GlobalEventHandlers/pointerup}} event.</p>
```

Comment:
I'm not sure if this is true, or even should be specified here. I.e. isn't it up the ui-events spec what interactions can cause these events? ui-events is specific to say it must be after a mousedown if triggered by a right mouse button: https://www.w3.org/TR/uievents/#event-type-click:~:text=When%20the%20contextmenu%20event%20is%20triggered%20by%20right%20mouse%20button%2C%20the%20contextmenu%20event%20MUST%20be%20dispatched%20after%20the%20mousedown%20event.

Also doesn't the section before say that events can be generated by a non-pointing device.. if so must these fire pointerdown/pointerup?

Maybe it would be better to say if ther user interaction which caused the firing of e is a pointerevent, t, then ...

Reply:
Done.

I left the platform convention as a non-normative text, with the thought that _some_ spec should mention this.  What do you think?